Macros/Compile_ACS.sas

```diff
@@ -125,20 +147,59 @@
     if inGeo;
     
     ** Create standard geography variable **;
-    
+
+	%if &geo = CITY %then %do;
+    length &geo_var $ 1;
+  
+    &geo_var = "1";
+  
+    label &geo_var = "&geo_label";
+  
+    format &geo_var &geo_format;
+	%end;
+
+
+	%if &geo = COUNTY %then %do;
+    length &geo_var $ 5;
+  
+    &geo_var = state || county ;
+  
+    label &geo_var = "&geo_label";
+  
+    format &geo_var &geo_format;
+	%end;
+
+	%if &geo = PLACE %then %do;
+    length &geo_var $ 7;
+  
+    &geo_var = state || place ;
+  
+    label &geo_var = "Regional places";
+  
+    format &geo_var &geo_format;
+	%end;
+
+
+    %else %do;
     length &geo_var $ &geo_length;
   
     &geo_var = substr( left( geoid ), 8 );
   
     label &geo_var = "&geo_label";
   
     format &geo_var &geo_format;
+	%end;
     
     ** Check for invalid geo variable values **;
+	** RP edited to only check for DC geographies **;
+
+	%if &state_ab = DC %then %do;
     
     if put( &geo_var, &geo_vformat ) = '' then do;
```

Comment:
I got an error at this point, triggered by the &geo_vformat macro variable invocation:

`MPRINT(COMPILE_ACS):   if put( RegPlace, ) = '' then do;`
...
`ERROR 22-322: Expecting a format name.`  

The value of geo_vformat is assigned on line 82. 

`%let geo_vformat = %sysfunc( putc( &geo, $geovfmt. ) );`

You need to define validation formats for the new geographies, and then add the names of those formats to the $geovfmt. format in the General library. $geovfmt. is created in the **D:\DCData\Libraries\General\Prog\Geo_level_formats.sas** program. (It's possible that you did this locally, but if the new format is not in the remote library then I can't access it.)